src/test/java/lambda/part3/exercise/Exercise1.java

```diff
@@ -0,0 +1,92 @@
+package lambda.part3.exercise;
+
+import lambda.data.Employee;
+import lambda.data.JobHistoryEntry;
+import lambda.data.Person;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+import static org.junit.Assert.assertEquals;
+
+@SuppressWarnings({"WeakerAccess", "unused"})
+public class Exercise1 {
+
+    @Test
+    public void mapEmployeesToLengthOfTheirFullNames() {
+        List<Employee> employees = getEmployees();
+        List<Integer> lengths = new ArrayList<>();
+
+        // TODO функция извлечения информации о человеке из объекта сотрудника personExtractor: Employee -> Person
+
+        Function<Employee, Person> personExtractor = Employee::getPerson;
+
+        // TODO функция извлечения полного имени из информации о человеке fullNameExtractor: Person -> String
+
+        Function<Person, String> fullNameExtractor = Person::getFullName;
+
+        // TODO функция извлечения длины из строки stringLengthExtractor: String -> Integer
+
+        Function<String, Integer> stringLengthExtractor = String::length;
+
+        // TODO функция извлечения длины полного имени из сотрудника fullNameLengthExtractor: Employee -> Integer
+
+        Function<Employee, Integer> fullNameLengthExtractor = employee -> stringLengthExtractor.apply(
+                fullNameExtractor.apply(
+                        personExtractor.apply(employee)));
```

Comment:
It's better to use `... .andThen(...).andThen(...)` because this is clearer.
For example, `extractPerson.andThen(extractFullName).andThen(extractLength)`.